src/tx/oracle.rs

```diff
@@ -16,7 +16,7 @@ pub enum CommitOutcome<E> {
 pub struct Oracle {
     pub(super) write_serialize_lock: Mutex<BTreeMap<u64, ConflictManager>>,
     pub(super) seqno: SequenceNumberCounter,
-    pub(super) snapshot_tracker: SnapshotTracker,
+    pub(super) snapshot_tracker: Arc<SnapshotTracker>,
```

Comment:
Is there a reason for changing the SnapshotTracker to require a manual Arc?

Reply:
The original SnapShotTracker struct was purely a struct around an `Arc`ed SnapShotTrackerInner and added no functionality, so I removed it to better see where `Arc`s were allocated. I did the same with CompactManager.

Reply:
While the Inner type has no functionality itself, it is itself Arc'd inside the outer struct, so when you clone it, you don't have to clone all its fields. So I think it's still useful to have.

Reply:
SnapShotTracker is only created once in KeyspaceInner inside an Arc. Itself (it has no Clone implementation) and its fields are never cloned. Oracle clones the Arc from Keyspace(Inner). I can revert the change but I found it as first time contributor more confusing to understand what the purpose of the wrapper structs were due to them behaving more like a type alias.

Reply:
Yeah right if it's wrapped in an Arc again it doesn't make a difference